shell/src/stats/memory.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use nix;
+use page_size;
+use serde::Serialize;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use std::error;
+use std::fmt;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::Path;
+use std::process::Command;
+use core::str::{from_utf8};
+
+pub type MemoryStatsResult<T> = std::result::Result<T, MemoryStatsError>;
+
+#[derive(Serialize, PartialEq, Clone, Debug)]
+pub struct LinuxData {
+    page_size: usize,
+    size: String,
+    resident: String,
+    shared: String,
+    text: String,
+    lib: String,
+    data: String,
+    dt: String
+}
+
+#[derive(Serialize, PartialEq, Clone, Debug)]
+pub struct DarwinOsData {
+    page_size: usize,
+    mem: f64,
+    resident: String
+}
+
+#[derive(Serialize,PartialEq, Clone, Debug)]
+#[serde(untagged)]
+pub enum MemoryData {
+    Linux(LinuxData),
+    DarwinOs(DarwinOsData)
+}
+
+impl From<LinuxData> for MemoryData {
+    fn from(data: LinuxData) -> Self {
+        MemoryData::Linux(data)
+    }
+}
+
+impl From<DarwinOsData> for MemoryData {
+    fn from(data: DarwinOsData) -> Self {
+        MemoryData::DarwinOs(data)
+    }
+}
+
+#[derive(PartialEq, Debug)]
+pub enum MemoryStatsError {
```

Comment:
Use [failure](https://github.com/rust-lang-nursery/failure) crate to express errors.